File: ChatClientCS/Data/SampleMainWindowViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ChatClientCS.ViewModels;
using ChatClientCS.Models;
using System.Collections.ObjectModel;
using System.IO;

namespace ChatClientCS.Data
{
    public class SampleMainWindowViewModel : ViewModelBase
    {
        private string _userName;
        public string UserName
        {
            get { return _userName; }
            set
            {
                _userName = value;
                OnPropertyChanged();
            }
        }

        private ObservableCollection<Participant> _participants = new ObservableCollection<Participant>();
        public ObservableCollection<Participant> Participants
        {
            get { return _participants; }
            set
            {
                _participants = value;
                OnPropertyChanged();
            }
        }

        private Participant _selectedParticipant;
        public Participant SelectedParticipant
        {
            get { return _selectedParticipant; }
            set
            {
                _selectedParticipant = value;
                if (SelectedParticipant.HasSentNewMessage) SelectedParticipant.HasSentNewMessage = false;
                OnPropertyChanged();
            }
        }

        public SampleMainWindowViewModel()
        {
            ObservableCollection<ChatMessage> someChatter = new ObservableCollection<ChatMessage>();
            someChatter.Add(new ChatMessage
            {
                Author = "Batman",
                Message = "What do you think about the Batmobile?",
                Time = DateTime.Now,
                IsOriginNative = true
            });
            someChatter.Add(new ChatMessage
            {
                Author = "Batman",
                Message = "Coolest superhero ride?",
                Time = DateTime.Now,
                IsOriginNative = true
            });
            someChatter.Add(new ChatMessage
            {
                Author = "Superman",
                Message = "Only if you don't have superpowers :P",
                Time = DateTime.Now
            });
            someChatter.Add(new ChatMessage
            {
                Author = "Batman",
                Message = "I'm rich. That's my superpower.",
                Time = DateTime.Now,
                IsOriginNative = true
            });
            someChatter.Add(new ChatMessage
            {
                Author = "Superman",
                Message = ":D Lorem Ipsum something blah blah blah blah blah blah blah blah. Lorem Ipsum something blah blah blah blah.",
                Time = DateTime.Now
            });
            someChatter.Add(new ChatMessage
            {
                Author = "Batman",
                Message = "I have no feelings",
                Time = DateTime.Now,
                IsOriginNative = true
            });
            someChatter.Add(new ChatMessage
            {
                Author = "Batman",
                Message = "How's Martha?",
                Time = DateTime.Now,
                IsOriginNative = true
            });

            Participants.Add(new Participant { Name = "Superman", Chatter = someChatter, IsTyping = true, IsLoggedIn = true });
            Participants.Add(new Participant { Name = "Wonder Woman", Chatter = someChatter, IsLoggedIn = false });
            Participants.Add(new Participant { Name = "Aquaman", Chatter = someChatter, HasSentNewMessage = true });
            Participants.Add(new Participant { Name = "Captain Canada", Chatter = someChatter, HasSentNewMessage = true });
            Participants.Add(new Participant { Name = "Iron Man", Chatter = someChatter, IsTyping = true });

            SelectedParticipant = Participants.First();
        }
    }
}

File: ChatClientVB/Data/SampleMainWindowViewModel.vb
Imports System.Collections.ObjectModel

Public Class SampleMainWindowViewModel
    Inherits ViewModelBase

    Private _userName As String
    Public Property UserName As String
        Get
            If _userName Is Nothing Then _userName = String.Empty
            Return _userName
        End Get
        Set(value As String)
            _userName = value
            OnPropertyChanged()
        End Set
    End Property

    Private _participants As New ObservableCollection(Of Participant)
    Public Property Participants As ObservableCollection(Of Participant)
        Get
            Return _participants
        End Get
        Set(value As ObservableCollection(Of Participant))
            _participants = value
            OnPropertyChanged()
        End Set
    End Property

    Private _selectedParticipant As Participant
    Public Property SelectedParticipant As Participant
        Get
            Return _selectedParticipant
        End Get
        Set(value As Participant)
            _selectedParticipant = value
            OnPropertyChanged()
        End Set
    End Property

    Public Sub New()
        Dim someChatter As New ObservableCollection(Of ChatMessage)
        someChatter.Add(New ChatMessage With {.Author = "Batman", .Message = "The Batmobile sucks",
                        .Time = DateTime.Now, .IsOriginNative = True})
        someChatter.Add(New ChatMessage With {.Author = "Superman", .Message = "It always has...",
                        .Time = DateTime.Now})
        someChatter.Add(New ChatMessage With {.Author = "Batman", .Message = "Really?! You never said so before.",
                        .Time = DateTime.Now, .IsOriginNative = True})
        someChatter.Add(New ChatMessage With {.Author = "Superman",
                        .Message = "Didn't want to hurt your feelings :D. Lorem Ipsum something blah blah blah blah blah blah blah blah.
                        Lorem Ipsum something blah blah blah blah.",
                        .Time = DateTime.Now})
        someChatter.Add(New ChatMessage With {.Author = "Batman", .Message = "I have no feelings",
                        .Time = DateTime.Now, .IsOriginNative = True})
        someChatter.Add(New ChatMessage With {.Author = "Batman", .Message = "How's Martha?",
                        .Time = DateTime.Now, .IsOriginNative = True})

        Participants.Add(New Participant With {.Name = "Superman", .Chatter = someChatter, .IsTyping = True})
        Participants.Add(New Participant With {.Name = "Wonder Woman", .Chatter = someChatter, .IsLoggedIn = False})
        Participants.Add(New Participant With {.Name = "Aquaman", .Chatter = someChatter, .HasSentNewMessage = True})
        Participants.Add(New Participant With {.Name = "Captain Canada", .Chatter = someChatter, .HasSentNewMessage = True})
        Participants.Add(New Participant With {.Name = "Iron Man", .Chatter = someChatter, .IsTyping = True})

        SelectedParticipant = Participants.First
    End Sub
End Class
